samples/clip-geometry/src/main/java/com/esri/arcgismaps/sample/clipgeometry/components/ClipGeometryViewModel.kt

```diff
@@ -0,0 +1,181 @@
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.symbology.SimpleFillSymbol
+import com.arcgismaps.mapping.symbology.SimpleFillSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleLineSymbol
+import com.arcgismaps.mapping.symbology.SimpleLineSymbolStyle
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.toolkit.geoviewcompose.MapViewProxy
+import com.esri.arcgismaps.sample.clipgeometry.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.launch
+
+class ClipGeometryViewModel(application: Application) : AndroidViewModel(application) {
+    val arcGISMap by mutableStateOf(
+        // create  a map with a topographic basemap style
+        ArcGISMap(BasemapStyle.ArcGISTopographic)
+    )
+
+    // create a MapViewProxy to interact with the MapView
+    val mapViewProxy = MapViewProxy()
+
+    // graphics overlay to display graphics
+    val graphicsOverlay = GraphicsOverlay()
+
+    private var coloradoGraphic  = createColoradoGraphic()
+    private var coloradoFillSymbol = SimpleFillSymbol(
+        SimpleFillSymbolStyle.Solid,
+        Color(R.color.transparentDarkBlue),
+        SimpleLineSymbol(SimpleLineSymbolStyle.Solid, Color.green, 2f)
+    )
+    private var envelopesGraphics = emptyList<Graphic>()
+
+    // Create a message dialog view model for handling error messages
+    val messageDialogVM = MessageDialogViewModel()
+
+    val isClipButtonEnabled = mutableStateOf(true)
+    val isResetButtonEnabled = mutableStateOf(false)
+
+    init {
+        viewModelScope.launch {
+            arcGISMap.load().onFailure { error ->
+                messageDialogVM.showMessageDialog(
+                    "Failed to load map",
+                    error.message.toString()
+                )
+            }.onSuccess {
+
+
+                // set the viewpoint of the map
+                mapViewProxy.setViewpoint(Viewpoint(40.0, -106.0, 10000000.0))
+
+                coloradoGraphic = createColoradoGraphic()
+                envelopesGraphics = createEnvelopeGraphics()
+
+                resetGeometry()
+
+            }
+        }
+    }
+
+
+    // clear the current graphics, then re-add the graphics for Colorado and the clip boxes
+    fun resetGeometry() {
+
+        graphicsOverlay.graphics.clear()
+
+        graphicsOverlay.graphics.add(coloradoGraphic)
+        coloradoGraphic.isVisible = true
+
+        envelopesGraphics.forEach { graphic ->
+            graphicsOverlay.graphics.add(graphic)
+        }
+
+        // update the reset button and clip button
+        isResetButtonEnabled.value = false
+        isClipButtonEnabled.value = true
+    }
+
+    // clip the Colorado geometry using the clip boxes and then add the result to the graphics overlay
```

Comment:
These function level comments should be in 

/ *
* Function level comments.
*/